src/main/webapp/map.js

```diff
@@ -56,19 +56,64 @@ var tempMarker = null;
 
 /**
  * Creates a new temp marker at the given coordinates with click and drag events
- * @param {google.maps.LatLng} latLng coordinates where to make the marker
+ * @param {google.maps.LatLng} coords coordinates where to make the marker
  */
-function makeTempMarker(latLng) {
-  tempMarker = new google.maps.Marker({position: latLng, map:map});
+function makeTempMarker(coords) {
+  tempMarker = new google.maps.Marker(
+    {
+      position: coords,
+      map: map,
+      label: "+"
+    });
   tempMarker.setDraggable(true);
 
   // clicking on a temp marker deletes it from the map
   tempMarker.addListener("click", () => {
-    tempMarker.setMap(null);
-    tempMarker = null;
+    makePermMarker(tempMarker.getPosition());
+    deleteTempMarker();
   });
-
   tempMarker.addListener('dragend', () => {
     map.panTo(tempMarker.getPosition());
   });
 }
+
+/** all permanent markers on the page */
+var permMarkers = new Set();
```

Comment:
Is it intentional that there cannot be two markers on the same location (I'm assuming that's the function of the Set in this case)

Reply:
This set only distinguishes between different Marker objects so no marker can be added here twice but you could have two different markers objects with the same coordinates. Although, you raise a good point because if you have two markers that are exactly on top of each other, you would not be able to access the one underneath without deleting the one on top which could be problematic.